sdk/azcore/etag.go

```diff
@@ -0,0 +1,47 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcore
+
+import (
+	"strings"
+)
+
+// ETag is a property used for optimistic concurrency during updates
+// ETag is a validator based on https://tools.ietf.org/html/rfc7232#section-2.3.2
+// An ETag can be empty ("").
+type ETag string
+
+// ETagAny is an ETag that represents everything, the value is "*"
+const ETagAny ETag = "*"
+
+// Equals does a strong comparison of two ETags. Equals returns true when both
+// ETags are not weak and the values of the underlying strings are equal. If both ETags are "nil" they are considered equal
+func (e ETag) Equals(other ETag) bool {
+	return !e.IsWeak() && !other.IsWeak() && e == other
+}
+
+// WeakEquals does a weak compariosn of two ETags. Two ETags are equivalent if their opaque-tags match
+// character-by-character, regardless of either or both being tagged as "weak". If both ETags are "nil" they are considered equal
+func (e ETag) WeakEquals(other ETag) bool {
```

Comment:
This doesn't handle the case of `nil` ETags per the comments.

Also, I'm leaning towards having the receivers passed by reference.  The reason being, if we are always defining ETags as `ETag *ETag` then calling this method will panic on a `nil` value.  If we make the receiver and parameter byref then we can perform the `nil` checks on the caller's behalf.  Thoughts?

Reply:
Fix spelling of "comparison". Remove doc sentence about "nil".
We don't want a pointer receiver because we want this to work like other fields in the struct where calling a method on nil panics - this is desired behavior.